nemesyst_core/mongo.py

```diff
@@ -591,6 +590,48 @@ def getBatches(self, db_batch_size=None, db_data_cursor=None):
                                   command_cursor.CommandCursor,
                                   "return": list}
 
+    def getFiles(self, db_batch_size=None, db_data_cursor=None,
+                 db_collection_name=None, db=None):
+        """Get gridfs files from mongodb by id using cursor to .files.
+
+        :param db_batch_size: The number of items to return in a single round.
+        :param db_data_cursor: The cursor to use to retrieve data from db.
+        :param db_collection_name: The top level collecton name
+            not including .chunks or .files where gridfs is to operate.
+        :param db: Database object to operate pipeline on.
+        :type db_batch_size: integer
+        :type db_data_cursor: command_cursor.CommandCursor
+        :type db_collection_name: str
+        :type db: pymongo.database.Database
+        :return: yields a list of tuples containing (item requested, metadata).
+        """
+        db_data_cursor = db_data_cursor if db_data_cursor is not None else \
+            self.args["db_data_cursor"]
+        db_batch_size = db_batch_size if db_batch_size is not None else \
+            self.args["db_batch_size"]
+        db_collection_name = db_collection_name if db_collection_name is not \
+            None else self.args["db_collection_name"]
+        db = db if db is not None else self.args["db"]
+
+        gfs = gridfs.GridFS(db, collection=db_collection_name)
```

Comment:
NOTE!!! this collection name is just the standard collection name. GridFS function expects just "example" not "example.chunks" or "example.files" but when creating the pipeline to highlight the data, this will indeed be "example.files".